### PR TITLE
Added a RAG chatbot with UI

### PR DESCRIPTION
**Pull request by tulu-g559**
📜 **Summary:**

*   Added a comprehensive README text for data extraction.
*   Included project overview, setup instructions, and usage guide.
*   Documented directory structure for better navigation.
*   Added contributing guidelines and licensing details.
    

🔧 **Changes Made:**

*   Created a well-structured README covering all project aspects.
*   Provided installation steps, environment setup, and execution guide.
*   Detailed contribution process for new contributors.
    

🚀 **Impact:**

*   Enhances project documentation for better accessibility.
*   Makes it easier for new contributors to set up and understand the project.
    

📌 **Review Notes:**

*   Check if any additional setup steps need inclusion.
*   Suggest improvements or formatting enhancements if needed.

#### Please review before taking any actions
Resolves Issue #23

### CODE REVIEW

**Comment by Hiteshydv001** [comment]
integrate with frontend

**Comment by Hiteshydv001** [comment]
can you add this with our vercel frontend

**Comment by tulu-g559** [comment]
@Hiteshydv001 
You can add another issue for that😭🙂

**Comment by Hiteshydv001** [comment]
give requirements.txt here in comments only

**Comment by Hiteshydv001** [comment]
freeze all the packages

**Comment by tulu-g559** [comment]
@Hiteshydv001 

here is the requirements

absl-py==2.1.0
aiohappyeyeballs==2.4.3
aiohttp==3.10.10
aiosignal==1.3.1
altair==5.4.1
annotated-types==0.7.0
anyio==4.4.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.1
astunparse==1.6.3
async-lru==2.0.4
attrs==24.2.0
babel==2.16.0
beautifulsoup4==4.12.3
bleach==6.1.0
blinker==1.8.2
CacheControl==0.14.2
cachetools==5.5.0
caer==2.0.8
certifi==2024.7.4
cffi==1.17.0
chardet==5.2.0
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
comm==0.2.2
contourpy==1.2.1
cryptography==44.0.0
cycler==0.12.1
dataclasses-json==0.6.7
debugpy==1.8.5
decorator==5.1.1
deepface==0.0.93
defusedxml==0.7.1
distro==1.9.0
dnspython==2.7.0
executing==2.0.1
faiss-cpu==1.9.0
fastapi==0.115.8
fastjsonschema==2.20.0
filelock==3.15.4
fire==0.6.0
firebase-admin==6.6.0
Flask==3.0.3
Flask-Cors==4.0.1
Flask-SQLAlchemy==3.1.1
flatbuffers==24.3.25
fonttools==4.53.1
fpdf==1.7.2
fqdn==1.5.1
frozenlist==1.5.0
fsspec==2024.10.0
gast==0.6.0
gdown==5.2.0
gitdb==4.0.11
GitPython==3.1.43
google-ai-generativelanguage==0.6.15
google-api-core==2.21.0
google-api-python-client==2.149.0
google-auth==2.35.0
google-auth-httplib2==0.2.0
google-cloud-core==2.4.1
google-cloud-firestore==2.20.0
google-cloud-storage==2.19.0
google-crc32c==1.6.0
google-generativeai==0.8.4
google-pasta==0.2.0
google-resumable-media==2.7.2
googleapis-common-protos==1.65.0
greenlet==3.1.1
grpcio==1.67.0
grpcio-status==1.67.0
gTTS==2.5.3
gunicorn==23.0.0
h11==0.14.0
h5py==3.11.0
httpcore==1.0.5
httplib2==0.22.0
httpx==0.27.0
httpx-sse==0.4.0
huggingface-hub==0.26.2
idna==3.7
ipykernel==6.29.5
ipython==8.26.0
ipywidgets==8.1.5
isoduration==20.11.0
itsdangerous==2.2.0
jax==0.4.31
jaxlib==0.4.31
jedi==0.19.1
Jinja2==3.1.4
jiter==0.7.0
joblib==1.4.2
json5==0.9.25
jsonpatch==1.33
jsonpointer==3.0.0
jsonschema==4.23.0
jsonschema-specifications==2023.12.1
jupyter==1.0.0
jupyter-console==6.6.3
jupyter-events==0.10.0
jupyter-lsp==2.2.5
jupyter_client==8.6.2
jupyter_core==5.7.2
jupyter_server==2.14.2
jupyter_server_terminals==0.5.3
jupyterlab==4.2.4
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.3
jupyterlab_widgets==3.0.13
keras==3.5.0
kiwisolver==1.4.5
langchain==0.3.7
langchain-community==0.3.5
langchain-core==0.3.15
langchain-google-genai==2.0.4
langchain-text-splitters==0.3.2
langsmith==0.1.140
libclang==18.1.1
loguru==0.7.3
lxml==4.9.4
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==2.1.5
marshmallow==3.23.1
matplotlib==3.9.2
matplotlib-inline==0.1.7
mdurl==0.1.2
mediapipe==0.10.14
mistune==3.0.2
ml-dtypes==0.4.0
mlxtend==0.23.1
msgpack==1.1.0
mtcnn==0.1.1
multidict==6.1.0
mypy==1.11.1
mypy-extensions==1.0.0
namex==0.0.8
narwhals==1.13.2
nbclient==0.10.0
nbconvert==7.16.4
nbformat==5.10.4
nest-asyncio==1.6.0
nltk==3.9.1
notebook==7.2.1
notebook_shim==0.2.4
numpy==1.26.4
openai==1.54.1
opencv-contrib-python==4.10.0.84
opencv-python==4.10.0.84
opencv-python-headless==4.10.0.84
opt-einsum==3.3.0
optree==0.12.1
orjson==3.10.11
outcome==1.3.0.post0
overrides==7.7.0
packaging==24.1
pandas==2.2.3
pandocfilters==1.5.1
parso==0.8.4
pillow==10.4.0
platformdirs==4.2.2
playsound==1.3.0
prometheus_client==0.20.0
prompt_toolkit==3.0.47
propcache==0.2.0
proto-plus==1.24.0
protobuf==5.28.2
psutil==6.0.0
pure_eval==0.2.3
pyarrow==18.0.0
pyasn1==0.6.1
pyasn1_modules==0.4.1
pycparser==2.22
pydantic==2.9.2
pydantic-settings==2.6.1
pydantic_core==2.23.4
pydeck==0.9.1
pygame==2.6.1
Pygments==2.18.0
PyJWT==2.10.1
pymongo==4.11.1
PyMuPDF==1.25.3
pyparsing==3.1.2
PyPDF2==3.0.1
PySocks==1.7.1
pytesseract==0.3.13
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-json-logger==2.0.7
python-telegram-bot==21.10
pytz==2024.1
pywin32==306
pywinpty==2.0.13
PyYAML==6.0.2
pyzmq==26.2.0
qtconsole==5.5.2
QtPy==2.4.1
referencing==0.35.1
regex==2024.9.11
reportlab==4.3.1
requests==2.32.3
requests-futures==1.0.2
requests-toolbelt==1.0.0
retina-face==0.0.17
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.7.1
rpds-py==0.20.0
rsa==4.9
safetensors==0.5.2
scikit-learn==1.5.1
scipy==1.14.1
seaborn==0.13.2
selenium==4.28.1
Send2Trash==1.8.3
setuptools==73.0.1
six==1.16.0
smmap==5.0.1
sniffio==1.3.1
sortedcontainers==2.4.0
sounddevice==0.5.0
soupsieve==2.6
SpeechRecognition==3.10.4
SQLAlchemy==2.0.35
stack-data==0.6.3
starlette==0.45.3
streamlit==1.39.0
tenacity==9.0.0
tensorboard==2.17.1
tensorboard-data-server==0.7.2
tensorflow==2.17.0
tensorflow-intel==2.17.0
termcolor==2.4.0
terminado==0.18.1
tf_keras==2.17.0
threadpoolctl==3.5.0
tinycss2==1.3.0
tokenizers==0.21.0
toml==0.10.2
tornado==6.4.1
tqdm==4.66.5
traitlets==5.14.3
transformers==4.49.0
trio==0.28.0
trio-websocket==0.11.1
types-python-dateutil==2.9.0.20240821
typing-inspect==0.9.0
typing_extensions==4.12.2
tzdata==2024.1
undetected-chromedriver==3.5.5
uri-template==1.3.0
uritemplate==4.1.1
urllib3==2.2.2
uvicorn==0.34.0
watchdog==5.0.3
wcwidth==0.2.13
webcolors==24.8.0
webencodings==0.5.1
websocket-client==1.8.0
websockets==14.2
Werkzeug==3.0.4
wheel==0.44.0
widgetsnbextension==4.0.13
win32_setctime==1.2.0
wrapt==1.16.0
wsproto==1.2.0
yahooquery==2.3.7
yarl==1.17.1

**Comment by Hiteshydv001** [comment]
bro, this code is not working at all also i can't find any logic for db of documents

**Comment by Hiteshydv001** [comment]
even api is also not working

**Comment by tulu-g559** [comment]
Let me look into it,,,
